<?xml version="1.0"?>
<launch>
    <!-- Launch arguments -->
    <arg name="dataset_dir" default="/home/agilex/data" doc="Directory to save captured data"/>
    <arg name="episode_index" default="0" doc="Episode index for this capture session"/>
    <arg name="type" default="aloha" doc="Type of data capture configuration"/>
    <arg name="config_path" default="" doc="Path to configuration YAML file"/>
    
    <!-- Data capture node -->
    <node pkg="data_tools" 
          type="data_capture_node.py" 
          name="data_capture_node" 
          output="screen">
        <param name="dataset_dir" value="$(arg dataset_dir)"/>
        <param name="episode_index" value="$(arg episode_index)"/>
        <param name="type" value="$(arg type)"/>
        <param name="config_path" value="$(arg config_path)"/>
    </node>
</launch>
